jina/executors/__init__.py

```diff
@@ -195,19 +195,18 @@ def workspace(self) -> str:
 
         :return: returns the workspace of the shard of this Executor.
         """
-        if getattr(self.runtime_args, 'workspace', None):
-            complete_workspace = os.path.join(
-                self.runtime_args.workspace, self.metas.name
-            )
+        workspace = getattr(self.metas, 'workspace') or getattr(
```

Comment:
```suggestion
        workspace = getattr(self.metas, 'workspace', or getattr(self.runtime_args, 'workspace', None))
```

Reply:
Good idea, but seems that it doesn't work.

The default workspace in metas is `''` which breaks the `getattr`

See https://github.com/jina-ai/jina/blob/c7e45999cc40fb220cea5fe3774d4156049ec67e/jina/executors/metas.py#L15

Not sure if this choice was intentional